packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2704,14 +2710,82 @@ class _Location {
   }
 }
 
+/// Describer to gather important information about [debugCreator].
+///
+/// This function will be attached to [FlutterErrorDetails.describeCreator]
+/// in [WidgetsBinding].
+Iterable<DiagnosticsNode> describeCreator(dynamic debugCreator) {
+  final Element element = debugCreator is Element? debugCreator: debugCreator.element;
+  return <DiagnosticsNode>[
+    _describeRelevantUserCode(element),
+  ].where((DiagnosticsNode node) => node != null);
+}
+
+DiagnosticsNode _describeRelevantUserCode(Element element) {
+  if (!WidgetInspectorService.instance.isWidgetCreationTracked()) {
+    return ErrorDescription(
+      'Widget creation tracking is currently disabled. Enabling'
+      ' it enables improved error messages. It can be enabled by passing '
+      '`--track-widget-creation` to `flutter run` or `flutter test`.',
+    );
+  }
+  DiagnosticsBlock desc;
+  element.visitAncestorElements((Element ancestor) {
+    // TODO(chunhtai): should print out all the widgets that are about to cross
+    // package boundaries.
+    if (_isLocalCreationLocation(ancestor)) {
+      desc = DiagnosticsBlock(
+        name: 'User created ancestor of the error causing widget was',
```

Comment:
this should say "error-causing".

The difference is that as written today, you're talking about the ancestor of the error, and saying that that ancestor is causing "widget". With the hyphen, you're saying that it's the ancestor of the widget causing the error.